openpype/lib/applications.py

```diff
@@ -1345,22 +1345,24 @@ def _prepare_last_workfile(data, workdir, workfile_template_key):
 
     # Last workfile path
     last_workfile_path = ""
-    extensions = avalon.api.HOST_WORKFILE_EXTENSIONS.get(
-        app.host_name
-    )
-    if extensions:
-        anatomy = data["anatomy"]
-        # Find last workfile
-        file_template = anatomy.templates[workfile_template_key]["file"]
-        workdir_data.update({
-            "version": 1,
-            "user": get_openpype_username(),
-            "ext": extensions[0]
-        })
-
-        last_workfile_path = avalon.api.last_workfile(
-            workdir, file_template, workdir_data, extensions, True
-        )
+    if not data.get("last_workfile_path"):  # to inject explicitly
+        extensions = avalon.api.HOST_WORKFILE_EXTENSIONS.get(app.host_name)
+
+        if extensions:
+            anatomy = data["anatomy"]
+            # Find last workfile
+            file_template = anatomy.templates["work"]["file"]
+            workdir_data.update({
+                "version": 1,
+                "user": get_openpype_username(),
+                "ext": extensions[0]
+            })
+
+            last_workfile_path = avalon.api.last_workfile(
+                workdir, file_template, workdir_data, extensions, True
+            )
+    else:
+        last_workfile_path = data.get("last_workfile_path")
```

Comment:
This is too complicated.
```suggestion
    last_workfile_path = data.get("last_workfile_path") or ""
    if not last_workfile_path:
        extensions = avalon.api.HOST_WORKFILE_EXTENSIONS.get(app.host_name)

        if extensions:
            anatomy = data["anatomy"]
            # Find last workfile
            file_template = anatomy.templates["work"]["file"]
            workdir_data.update({
                "version": 1,
                "user": get_openpype_username(),
                "ext": extensions[0]
            })

            last_workfile_path = avalon.api.last_workfile(
                workdir, file_template, workdir_data, extensions, True
            )
```